sway-fmt-v2/src/items/item_enum.rs

```diff
@@ -182,9 +182,11 @@ fn format_enum(
 
 impl ItemLen for ItemEnum {
     fn get_formatted_len(&self) -> Result<usize, FormatterError> {
-        // TODO while determininig the length we may want to format to some degree and take length.
-        let str_item = &self.span().as_str().len();
-        Ok(*str_item as usize)
+        // Format to single line and return the length
+        let mut str_item = String::new();
+        let mut formatter = Formatter::default();
+        format_enum(self, &mut str_item, &mut formatter, false)?;
+        Ok(str_item.len() as usize)
```

Comment:
We likely want to use `chars().count()` rather than `len` here. The `len` method gives us the length in _bytes_, but in UTF-8 (which is what the `str` and `String` types represent) its possible for a single character to occupy more than one byte. The [str::len docs](https://doc.rust-lang.org/std/primitive.str.html#method.len) briefly mention this.

Using `str_item.chars().count()` will give the length in UTF-8 characters which is closer to what we're talking about when we're checking things like "line width".

We might want to consider changing the `ItemLen` trait to `ItemLenChars` and change its method to `len_chars` to make this a bit clearer. We should review the rest of the formatter code-base for uses of `str::len` too where this might apply.

Apologies for not spotting this sooner on the other implementations!

Reply:
Ah, I also encountered this my last run in with the old formatter. I should've noticed as well.

Reply:
Ohh, I see. I am changing it rn and will look into the existing code base for other uses of `str::len` 👍